### PR TITLE
Product Collection: break taxonomies filter into multiple menu items per each taxonomy

### LINKED ISSUE #42049: Product Collection - Break Taxonomies filter into multiple menu items
We've discussed(p1700653504142969-slack-C05UMTA7W0P) restructuring the "Taxonomies" filters in the Product Collection block. The current setup combines all taxonomies under a single filter, which can be cumbersome and less intuitive for users.

### **Proposed Changes**

The proposal is to divide the "Taxonomies" filter into multiple filters. These would include:

1. **Product Categories**: A dedicated filter for product categories.
2. **Product Tags**: A separate filter for product tags.
3. **Custom Taxonomies**: Individual filters for each custom taxonomy, such as "Brands".

### PR DESCRIPTION
**Pull request by kmanijak**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

In Product Collection block, merchants can filter products by taxonomies. Till now it was a single "Taxonomies" entry in a filter that opened multiple inputs per taxonomy. It's unintuitive.

This PR splits the "Taxonomies" filter into separate filters per taxonomy (by default: "Product categories" and "Product Tags" but also custom taxonomies).

Closes https://github.com/woocommerce/woocommerce/issues/42049

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

#### Replaced "Taxonomies" with specific taxonomies filters

1. To your `functions.php` add the following code. This code registers custom taxonomy.

```
function add_taxonomy() {
	register_taxonomy( 'custom-product-taxonomy', array( 'product' ), array(
		'public'             => true,
		'publicly_queryable' => true,
		'show_in_rest' 		   => true,
		'label'              => 'Custom Taxonomy e.g. Brand',
	) );
}
add_action( 'init', 'add_taxonomy' );
```

2. Create new post
3. Insert Product Collection
4. Choose Product Catalog collection
5. Open Inspector Controls and click Filters three-dot menu
6. **Expected**: There's no "Taxonomies" entry but:
    - "Product Categories"
    - "Product Tags"
    - "Custom Taxonomy e.g. Brand"

Before | After
-- | --
Screen Shot 2024-04-03 at 11 16 32 AM | Screen Shot 2024-04-03 at 11 14 10 AM

#### Filter works

1. Continue from the previous scenario - choose "Product Categories"
2. Type and choose some category you have in store, e.g. "Accessories"
3. **Expected**: products are filtered by Category
4. Open Filters three-dot menu
5. **Expected**: make sure "Product Category" and only this one from taxonomies filters has the tick next to it

Screen Shot 2024-04-03 at 11 20 20 AM

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/taxonomy-controls/index.tsx [hunk 1]
```diff
@@ -74,15 +65,22 @@ function TaxonomyControls( {
...
-					<TaxonomyItem
+					<ToolsPanelItem
 						key={ taxonomy.slug }
-						taxonomy={ taxonomy }
-						termIds={ termIds }
-						onChange={ handleChange }
-					/>
+						label={ taxonomy.name }
+						hasValue={ () => termIds.length }
+						onDeselect={ () => handleChange( [] ) }
```
Comment (imanish003) [inline plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/taxonomy-controls/index.tsx]: As implemented in [this pull request](https://github.com/woocommerce/woocommerce/pull/46226), do we also need to add 'resetAllFilter' for this?

**Review by imanish003 — APPROVED** [review]
Good job! I tested the changes & everything is working as expected. I just left a minor suggestion but pre-approving the PR as I don't consider that as a blocker 🚀

File: plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/taxonomy-controls/index.tsx (hunk 1 above)
Comment (kmanijak) [reply plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/taxonomy-controls/index.tsx]: There will be conflict anyway, so depending on which one I manage to merge first, I'll adjust the other PR (either add `resetAllFilter` here or split the reset in another one).

Comment (kmanijak) [reply plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/taxonomy-controls/index.tsx]: Conflict resolved and `resetAllFilter` is added to each taxonomy separately 🙌

**Pull request merged**